### DISCUSSION TITLE
Make fixes/improvements to map window

### DISCUSSION THREAD

**Discussion by tellovishous** [OpenRCT2/OpenRCT2#15744] [Feature requests]
Wanted to make this post because looking at the Map window, there's a number of different things that can be done to improve it and I feel like it's been a tad neglected compared to other aspects of the ORCT2 project thus far, so I thought I'd try consolidating the issues it currently has and enhancements that could be made to it:

- The minimap is not properly centered, cuts off corners on some rotations, and has excessive blank space around it. (original bug) (#7248 , #13294)
- Add a zoom feature to the minimap to make it and its elements easier to see. (#2782 , #3785 , #5410)
- Add filtering options to select which elements show up in the map. (#3785)

And a few more suggestions of my own:
- Have the window open in a bigger size so that it can show the whole map on first load and doesn't need to be dragged out bigger.
- When selecting the "Show rides and stalls on map" tab, have the color references expand downward from the window, instead of upward inside the window where it can block the map view.
- Perhaps look into uncapping the rendering of the map so it all shows up quicker instead of the left-downward swipe it currently makes.

There might be some other map issues I haven't mentioned here but I wanted to list the more prominent ones. Feel free to discuss this and any other issues/additions we could make.

**Comment by ocalhoun6** [comment]
- Maybe it would be possible to auto-zoom the map to fit the current dimensions of the map window? So if you want to 'zoom in' your mini-map, you just make the window bigger. And since the map would always fit perfectly in the window, you no longer need scroll bars.

- Crazy idea: putting a diamond-shaped map in a rectangular window is always going to result in a good bit of wasted screen space. _But_... What if the window didn't have to be rectangular? Yeah, the top of it probably needs to stay rectangular to fit the tabs, window name, close button, etc ... but the bottom corners of the window could be possible to cut off. Less crazy version: those empty corners of the window might be a good place to put the filtering buttons without needing to make the window bigger.
